src/utilities/getFilteredNotes.js

```diff
@@ -0,0 +1,52 @@
+export const getFilteredNotes = (state, notesList) => {
+  //Filter By Priority
+  if (state.filterByPriority.length > 0) {
+    let dummyList = [];
+    state.filterByPriority.map(
+      (priorityName) =>
+        (dummyList = dummyList.concat(
+          notesList.filter((note) => note.priority.includes(priorityName))
+        ))
+    );
+    notesList = [...dummyList];
+  }
+  //Filter By Labels
+  if (state.filterByLabels.length > 0) {
+    let dummyList = [];
+    state.filterByLabels.map(
+      (labelsName) =>
+        (dummyList = dummyList.concat(
+          notesList.filter((note) => note.labelsData.includes(labelsName))
+        ))
+    );
+    notesList = [...dummyList];
+  }
+
+  //Sort by Date
+  if (state.sortByDate === "newestFirst") {
+    //newest first
+    notesList = notesList.sort(
+      (note1, note2) =>
+        new Date(note2.noteCreatedDate) - new Date(note1.noteCreatedDate)
+    );
+  }
+  if (state.sortByDate === "oldestFirst") {
+    notesList = notesList.sort(
+      (note1, note2) =>
+        new Date(note1.noteCreatedDate) - new Date(note2.noteCreatedDate)
+    );
+  }
+
+  let highNotes = notesList.filter((note) => note.priority == "high");
+  let mediumNotes = notesList.filter((note) => note.priority == "medium");
+  let lowNotes = notesList.filter((note) => note.priority == "low");
+  let noneNotes = notesList.filter((note) => note.priority == "none");
+  if (state.sortByPriority === "HighToLow") {
+    notesList = [...highNotes, ...mediumNotes, ...lowNotes, ...noneNotes];
+  }
+  if (state.sortByPriority === "LowToHigh") {
+    notesList = [...noneNotes, ...lowNotes, ...mediumNotes, ...highNotes];
```

Comment:
What exactly this piece of code is doing? Please explain, I am unable to understand maybe because of less knowledge.

Reply:
Np Paarth, First I have filtered out notes with respective priorities into variables,
Then I have used spread operators in array to merge all notes in respective order [high to none] priority & vice versa.
There may be good way of doing this but due to less time, I have implemented this.